### DISCUSSION TITLE
Section 6: 34 - problema na criação da lambda

### DISCUSSION THREAD

**Discussion by Alves0611** [chgasparoto/curso-aws-com-terraform#138] [Q&A]
estou recebendo essa mensagem quando aplico o apply, sabe me dizer por que?

Error: filename, s3_* or image_uri attributes must be set

  with aws_lambda_function.s3,
  on lambda.tf line 1, in resource "aws_lambda_function" "s3":
    1: resource "aws_lambda_function" "s3" {

**Comment by chgasparoto** [comment]
@Alves0611 verifique se você está especificando uma das três propriedades requiridas. As que estão descritas na mensagem de erro.
No curso utilizamos a propriedade `filename`.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#filename

**Reply by Alves0611** [reply]
bom dia @chgasparoto estou especificando  


resource "aws_lambda_function" "s3" {
  function_name = "s3"
  handler       = "index.handler"
  role          = aws_iam_role.s3.arn
  runtime       = "nodejs14.x"

}



agora estou recebendo esse erro 


Error: expected runtime to be one of [nodejs nodejs4.3 nodejs6.10 nodejs8.10 nodejs10.x nodejs12.x java8 java8.al2 java11 python2.7 python3.6 python3.7 python3.8 dotnetcore1.0 dotnetcore2.0 dotnetcore2.1 dotnetcore3.1 nodejs4.3-edge go1.x ruby2.5 ruby2.7 provided provided.al2], got nodejs14.x
│ 
│   with aws_lambda_function.s3,
│   on lambda.tf line 5, in resource "aws_lambda_function" "s3":
│    5:   runtime       = "nodejs14.x"




sabe me dizer o que está acontecendo?  vi na documentação que os argumentos obrigatorios são "function_name" e "role"

**Reply by chgasparoto** [reply]
qual versão do aws provider você está utilizando? Como você pode ver na lista, não tem a versão 14 do node na mensagem de erro.
